### PR TITLE
Update online documentation

### PR DESCRIPTION
**Pull request by yardasol**
## Summary of changes

This PR adds @mehmeturkmen's changes to the docs on `master` to the website.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Required for Merging
- [x] I have read the **CONTRIBUTING** document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
  - [ ] CI tests pass
  - [ ] Local tests pass (including Serpent2 integration tests)

## Associated Issues and PRs

- Issue: #58

## Associated Developers

- Dev: @munkm 

## Checklist for Reviewers

Reviewers should use [this link](https://arfc.github.io/manual/guides/pull_requests) to get to the 
Review Checklist before they begin their review.

### CODE REVIEW

**Comment by LukeSeifert** [comment]
Do you have a list of the files that include the updated changes to the docs from Mehmet's changes? I spent a bit of time looking through the "files changed", but I can't seem to find the documentation changes since there are a lot of changed files. I also tried checking the previous docs (https://arfc.github.io/saltproc/) and comparing with your build with github-pages, but I couldn't seem to find the documentation differences since I didn't know where they would be on the site.

File: _sources/src/saltproc.rst.txt [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
-   saltproc.version
-
-Module contents
----------------
-
-.. automodule:: saltproc
-   :members:
-   :undoc-members:
-   :show-inheritance:
+saltproc package
```
Comment (munkm) [inline _sources/src/saltproc.rst.txt:1]: how did this file change? I can't see the difference here.

File: _static/basic.css [hunk 2]
```diff
@@ -4,7 +4,7 @@
  *
```
Comment (munkm) [inline _static/basic.css:4]: did the styling get updated by memo or should that be a different PR?

**Review by munkm — COMMENTED** [review]
Hey, it seems like a lot more than generated HTML are in this PR (e.g. all the `_static/fonts/`) Do you know why that is?

File: _static/basic.css (hunk 2 above)
Comment (mehmeturkmen) [reply _static/basic.css:4]: Hi @munkm and @yardasol . I saw your comments on this PR. it seems you are having an issue to figure out the differences from the previous PR. So, let me explain. What you are looking at is gh-pages containing html and supporting documents for https://arfc.github.io/saltproc/. I updated these files in a separate PR. You need to create all these files all over again when you change any file under saltproc/doc/ by using Sphinx. So, you may not be able to see the difference by directly looking into files in gh-pages. For your convenience, I put below the update I made in that PR,
In "overview.html", 

Updates
March 2021:

Besides the existing flexibility like fixed removal efficiency definition for each target isotope defined in the object input file, Saltproc code now comprises Sparging System package that calculates removal efficiencies for various target isotopes (i.e., Xe, Kr, and H). To enable this feature, use the “self” command in the input file in the “efficiency” object names of Sparger and Separator components. Each component can be employed separately. An example is given below. In the example, capacity and mass flow rate are in g/s unit while volume is in cm3 unit.

“sparger”: { “capacity”: 9920000,
“efficiency”: “self”, “mass_flowrate”: 9920000, “volume”: 10000000 },

“entrainment_separator”: { “capacity”: 9920000,
“efficiency”: “self”, “mass_flowrate”: 9920000, “volume”: 11 }

Please for all changes I made, take a look at my last PR in master, not in gh-pages.

File: _sources/src/saltproc.rst.txt (hunk 1 above)
Comment (yardasol) [reply _sources/src/saltproc.rst.txt:1]: > Hey, it seems like a lot more than generated HTML are in this PR (e.g. all the `_static/fonts/`) Do you know why that is?

We should add a guide somewhere on this in the near future, but for now, here's my understanding of what's going on here based on the instructions in the read me for building the documentation. The TL;DR is that the entire website gets rebuilt every time we update the documentation. I think @mehmeturkmen alluded to this in his comment. Below is the full version, for those interested.

0. We first need to build and install saltproc. If we have done this in an environment, we need to make sure that environment is active before we go build the docs. This is so `sphinc-apidoc` can read the different internal (`Depcode`, `App`, etc) and external (`pyne`, ` modules 

1. running `sphinx-apidoc --separate --force --output-dir=src/ ../saltproc` pulls the docstrings from the source code and places them in formatted `rst` files under `saltproc/src/`:
```
Creating file src/setup.rst.
Creating file src/saltproc.rst.
Creating file src/saltproc.app.rst.
Creating file src/saltproc.depcode.rst.
Creating file src/saltproc.materialflow.rst.
Creating file src/saltproc.process.rst.
Creating file src/saltproc.reactor.rst.
Creating file src/saltproc.separator.rst.
Creating file src/saltproc.simulation.rst.
Creating file src/saltproc.sparger.rst.
Creating file src/saltproc.version.rst.
Creating file src/saltproc.tests.rst.
Creating file src/saltproc.tests.test_app.rst.
Creating file src/saltproc.tests.test_depcode.rst.
Creating file src/saltproc.tests.test_materialflow.rst.
Creating file src/saltproc.tests.test_process.rst.
Creating file src/saltproc.tests.test_separator.rst.
Creating file src/saltproc.tests.test_simulation.rst.
Creating file src/saltproc.tests.test_sparger.rst.
Creating file src/modules.rst.
```
2. Running `make clean` removes any previous `_build` folder that would contain all of the files that need to go onto `gh-pages`:
```
rm -rf _build/*
```

3. Running `make html` builds the files for the web based documentation. Notice that there are several warning below due to my environment being active during the build process. This shouldn't effect the website files (I think)
```
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v4.2.0
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 24 source files that are out of date
updating environment: [new config] 24 added, 0 changed, 0 removed
reading sources... [100%] src/saltproc.version                                          
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/separator.py:docstring of saltproc.separator.Separator.calc_rem_efficiency:1: WARNING: duplicate object description of saltproc.separator.Separator.calc_rem_efficiency, other instance in src/saltproc.separator, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/separator.py:docstring of saltproc.separator.Separator.description:1: WARNING: duplicate object description of saltproc.separator.Separator.description, other instance in src/saltproc.separator, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/separator.py:docstring of saltproc.separator.Separator.eff:1: WARNING: duplicate object description of saltproc.separator.Separator.eff, other instance in src/saltproc.separator, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/separator.py:docstring of saltproc.separator.Separator.k:1: WARNING: duplicate object description of saltproc.separator.Separator.k, other instance in src/saltproc.separator, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.calc_rem_efficiency:1: WARNING: duplicate object description of saltproc.sparger.Sparger.calc_rem_efficiency, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.description:1: WARNING: duplicate object description of saltproc.sparger.Sparger.description, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.diffusivity:1: WARNING: duplicate object description of saltproc.sparger.Sparger.diffusivity, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.eff:1: WARNING: duplicate object description of saltproc.sparger.Sparger.eff, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.eff:10: WARNING: Unexpected indentation.
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.eff:11: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.eps:1: WARNING: duplicate object description of saltproc.sparger.Sparger.eps, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.exp_const:1: WARNING: duplicate object description of saltproc.sparger.Sparger.exp_const, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.gas_const:1: WARNING: duplicate object description of saltproc.sparger.Sparger.gas_const, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.h_const:1: WARNING: duplicate object description of saltproc.sparger.Sparger.h_const, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.sherwood:1: WARNING: duplicate object description of saltproc.sparger.Sparger.sherwood, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.temp_room:1: WARNING: duplicate object description of saltproc.sparger.Sparger.temp_room, other instance in src/saltproc.sparger, use :noindex: for one of them
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/ooblack/projects/saltproc/doc/src/modules.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] src/saltproc.version                                   
generating indices... genindex py-modindex done
highlighting module code... [100%] saltproc.tests.test_sparger                  
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 17 warnings.

The HTML pages are in _build/html.

Build finished. The HTML pages are in _build/html.
``` 
This is probably is why there are so many files changed.

**Comment by yardasol** [comment]
> Do you have a list of the files that include the updated changes to the docs from Mehmet's changes? I spent a bit of time looking through the "files changed", but I can't seem to find the documentation changes since there are a lot of changed files. I also tried checking the previous docs (https://arfc.github.io/saltproc/) and comparing with your build with github-pages, but I couldn't seem to find the documentation differences since I didn't know where they would be on the site.

I think you are right that the docs are not different at all. On July 23, 2020, PR #54, #53, and #52 containing @mehmeturkmen's changes were merged into `master`, `master` and `gh-pages` respectively. There haven't really been any changes to the source code that need documentation changes since then.

@andrewryh were there any changes that were missed in #52 that prompted you to write #58, or was that just an exercise for me to learn how to do it?

If the latter, I think this is PR can be closed without merging.

**Comment by yardasol** [comment]
I will close this PR and make a new one for the upcoming 0.3.0 release that incorporates some new docstrings and code structure.
